theano/tensor/opt.py

```diff
@@ -4760,6 +4760,10 @@ def local_useless_elemwise_comparison(node):
     Elemwise[LT](add([anything that is shapes]), 0) -> Elemwise[zeros](X)
     Elemwise[GE](add([anything that is shapes]), 0) -> Elemwise[ones](X)
 
+    # Shapes are never negativ
```

Comment:
I'm also wondering if we should make shapes into unsigned ints so that all the special casing done here would be standard.